src/main/java/org/candlepin/pki/impl/JSSProviderLoader.java

```diff
@@ -109,7 +61,7 @@ public static String getJSSVersion() {
      * Initializes the JSS CryptoManager (which in turn initializes the NSS DB), and loads the JSS provider
      * into the JVM. The method should be called during context initialization. Can be called more than once
      * without ill effect (-1 will be returned if the provider is already installed).
-     *
+     * <p>
```

Comment:
Need to close the paragraph tag

Reply:
Javadoc uses only opening tags for paragraphs. But this is something I wanted to bring up. Some of our javadoc is not formatted correctly (such as paragraphs as javadoc ignores blank lines) and because of that the output gets squished into a single wall of text which makes documentation pop-ups in IDE less useful.

Reply:
+1 to addressing of paragraph formatting

The part I was mentioning here has something to do with how some parsers handle that dangling paragraph tag. I don't recall the details, and it may not be important (anymore?), but a number of years ago I remember we had to stop using the XHTML variant of the tag because it broke something or someone's IDE.

If you find anything on that which lines up with the memory fragment I have here, it'd be good to get a refresher as to what that issue is and whether or not we need to care about well-formed HTML in our docs.

Reply:
I checked how it works and it's a mess. Attempt to generate our javadoc results in ~100 errors and ~100 warnings. XHTML tags such as <br /> break it. Empty tags such as <p></p> break it and multitude of other issues.

I also found the JEP about HTML5, but unfortunately that makes only the output HTML5 compliant not the javadoc itself. But the solution is on the way. It looks like OpenJDK is working on supporting markdown in javadoc.

Reply:
It's both amusing and frustrating that the only syntax that works correctly is the one that's invalid and only in existence due to the legacy of everyone doing it wrong at the dawn of the web.

But good to know, and the move to markdown would be a welcome one -- even though we know we'll get yet another weird implementation that follows none of the existing standards.

Thanks for looking into this.